index.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+// const fs = require('fs');
+// const expect = require('chai').expect;
+const readfiles = require('./lib/read-files.js');
+const main = module.exports = function (argv){
+    if(argv.length !=5) return 'Please enter at least 3 arguments';
+        return readfiles(argv.slice(2,5), function(err, data){
+            if (err) {
+                return console.error(err);
+            }
+            return console.log(data);
+    })
+}
+
+console.log(main(process.argv));
```

Comment:
Just a small thing I wasn't aware of until i ran the code on my machine. Let's get rid of the ```console.log``` and just invoke the ```main``` function so that it says ```main(process.argv)```. 

When I ran your code in my terminal, I was first getting ```undefined``` and *then* the array of text output. If we just get rid of ```console.log``` and invoke ```main(process.argv)``` alone, we still get text output printed out because we already log the data on Line 12. 

The reason  for that is the console isn't able to log the output from ```main(process.argv)``` because the ```main``` function is async. 

In other words, ```console.log``` happens instantly while ```main(process.argv)``` is still going through the process of completing your ```readfiles``` function that happens on Line 8. 

```Console.log``` isn't going to wait until ```main(process.argv)``` is done, it's going to just print out contents right away, but since ```readFiles``` on Line 8 still hasn't completed, ```console.log``` spits out ```undefined.```

If you're confused, come find me and I'll try to re-explain again. :)